libraries/AP_InertialSensor/AP_InertialSensor_SITL.cpp

```diff
@@ -195,62 +195,55 @@ void AP_InertialSensor_SITL::generate_gyro()
     Vector3f gyro_accum;
     uint8_t nsamples = enable_fast_sampling(gyro_instance) ? 8 : 1;
 
+    const float _gyro_drift = gyro_drift();
     for (uint8_t j = 0; j < nsamples; j++) {
-        float p = radians(sitl->state.rollRate) + gyro_drift();
-        float q = radians(sitl->state.pitchRate) + gyro_drift();
-        float r = radians(sitl->state.yawRate) + gyro_drift();
+        float p = radians(sitl->state.rollRate) + _gyro_drift;
+        float q = radians(sitl->state.pitchRate) + _gyro_drift;
+        float r = radians(sitl->state.yawRate) + _gyro_drift;
 
         // minimum gyro noise is less than 1 bit
-        float gyro_noise = ToRad(0.04f);
-        float noise_variation = 0.05f;
+        constexpr float gyro_noise = ToRad(0.04f);
+        constexpr float noise_variation = 0.05f;
         // this smears the individual motor peaks somewhat emulating physical motors
-        float freq_variation = 0.12f;
+        constexpr float freq_variation = 0.12f;
         // add in sensor noise
         p += gyro_noise * rand_float();
         q += gyro_noise * rand_float();
         r += gyro_noise * rand_float();
 
-        bool motors_on = sitl->throttle > sitl->ins_noise_throttle_min;
         // on a real 180mm copter gyro noise varies between 0.2-0.4 rad/s for throttle 0.2-0.8
         // giving a gyro noise variation of 0.33 rad/s or 20deg/s over the full throttle range
-        if (motors_on) {
+        if (sitl->throttle > sitl->ins_noise_throttle_min) {
             // add extra noise when the motors are on
-            gyro_noise = ToRad(sitl->gyro_noise[gyro_instance]) * sitl->throttle;
-        }
+            const float motor_gyro_noise = ToRad(sitl->gyro_noise[gyro_instance]) * sitl->throttle;
 
-        // VIB_FREQ is a static vibration applied to each axis
-        const Vector3f &vibe_freq = sitl->vibe_freq;
-
-        if (vibe_freq.is_zero() && is_zero(sitl->vibe_motor)) {
-            // no rpm noise, so add in background noise if any
```

Comment:
Where did this bit go? I'm having a hard time following the diff

Reply:
It's a tough diff to follow.  But I see no reason we should be adding the gyro sensor noise *twice* - we've already added it on line 208.

Reply:
I think its different because gyro_noise would have been modified by the motors being on. i.e. what was added in original line 218 would have been added here.

Reply:
Yes, I agree it's different.

I don't understand the intent here, though.  The comment is "add in background noise" - the `ToRad(0.04f)` is to do with random noise in the sensor (I'm assuming that's what "background noise" and "sensor noise" mean in this function) which should be unrelated to what the current throttle level is.  So adding this sensor noise * throttle value in here is strange.

Should there be some other constant to represent whatever this component is supposed to be?